src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -515,6 +524,33 @@ object ICustomAstVisitor.VisitArrayLiteral(ArrayLiteralAst arrayLiteralAst)
 
         object ICustomAstVisitor.VisitHashtable(HashtableAst hashtableAst)
         {
+            if (hashtableAst.KeyValuePairs.Count > 0)
+            {
+                var properties = new List<PSMemberNameAndType>();
+
+                foreach (var kv in hashtableAst.KeyValuePairs)
+                {
+                    if (SafeExprEvaluator.TrySafeEval(kv.Item1, _context.ExecutionContext, out object nameValue))
```

Comment:
Shall we check for `StringConstantExpressionAst` first? There is a good chance we can avoid evaluating the expression.

Reply:
Yes, we should :) 
And probably check that for the value too. Chances are that it's either a number or a string, right?

Reply:
Yes, the value could also be a constant. Maybe check for `ConstantExpressionAst` for value?